homeassistant/components/logbook/__init__.py

```diff
@@ -466,6 +469,9 @@ def yield_events(query):
                     filters.entity_filter() | (Events.event_type != EVENT_STATE_CHANGED)
                 )
 
+        if context_id is not None:
```

Comment:
Can we make this mutually exclusive of passing `entity_ids` and put it in the else block?

If not it would be more efficient to write it into the two queries that are unioned in the `if entity_ids is not None:` so its filtered before the union

Reply:
I now added a check to block combining context_id with entity_ids. Not sure where the union is.

Reply:
The `union` is in the `if` block for passing `entity_ids` and is only used when selecting by `entity_id` since `events` doesn't have an index by entity id.